src/main/java/com/javaproject/restservice/SecurityConfig.java

```diff
@@ -0,0 +1,42 @@
+package com.javaproject.restservice;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.security.config.annotation.authentication.builders.AuthenticationManagerBuilder;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+import org.springframework.security.core.authority.AuthorityUtils;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+
+
+import com.javaproject.restservice.entity.User;
+import com.javaproject.restservice.repository.UserRepository;
+
+import java.util.*;
+
+@Configuration
+@EnableWebSecurity
+public class SecurityConfig extends WebSecurityConfigurerAdapter {
```

Comment:
Oooohhh. This just got interesting!
What is WebSecurityAdapter? What is and adapter?